hudi-common/src/main/avro/HoodieClusteringGroup.avsc

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+{
+   "namespace":"org.apache.hudi.avro.model",
+   "type":"record",
+   "name":"HoodieClusteringGroup",
+   "type":"record",
+   "fields":[
+      {
+         /* Group of files that needs to merged. All the slices in a group will belong to same partition initially.
```

Comment:
@satishkotha AVRO has a way to add description with a type called "doc", we can either use that or add these comments at the beginning of the avro file.

Reply:
@satishkotha Also, can we not create multiple avsc files for each "Pojo" that you need, you can just define them all under the HoodieClusteringPlan,  take a look at this for example -> https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCleanerPlan.avsc#L78. Once the code-gen creates the POJO from the AVSC, you will have access to them.

Reply:
From earlier reviews, creating multiple avsc was preferred because we can reuse pojos for other operations. Please see discussion in #2048 . Let me know if there are any advantages of keeping them together in one file

Reply:
Hmm, I see. If there are use-cases where we plan to reuse them to create different POJO's, I'm fine with this.